pkg/config/data.go

```diff
@@ -270,17 +270,17 @@ func (c *Credential) MarshalJSON() ([]byte, error) {
 // MarshalJSON omits empty structs
 func (srv *Server) MarshalJSON() ([]byte, error) {
 	type S Server
-	as := &srv.Authorization
+	az := &srv.Authorization
 	if (AuthorizationServer{}) == srv.Authorization {
-		as = nil
+		az = nil
 	}
-	rss := &srv.RedSky
-	if (RedSkyServer{}) == srv.RedSky {
-		rss = nil
+	api := &srv.API
+	if (APIServer{}) == srv.API {
+		api = nil
 	}
 	return json.Marshal(&struct {
 		*S
 		Authorization *AuthorizationServer `json:"authorization,omitempty"`
-		RedSky        *RedSkyServer        `json:"redsky,omitempty"`
-	}{S: (*S)(srv), Authorization: as, RedSky: rss})
+		API           *APIServer           `json:"api,omitempty"`
+	}{S: (*S)(srv), Authorization: az, API: api})
```

Comment:
True story. I need to go back and double check in the code, but I think it came down to the semantics of how they are referenced: I think I originally had concerns about what would happen to the rest of the code if these were allowed to be `nil` and it came down to where I wanted the ugly to live. It would be easier if something like the `IsZero` function were supported in the JSON encoder.